Document/0x06d-Testing-Data-Storage.md

```diff
@@ -315,55 +299,124 @@ $ grep -iRn keyword .
 
 Then you can monitor and verify the changes in the filesystem of the app and investigate if any sensitive information is stored within the files while using the app.
 
-##### Dynamic Analysis with Needle
+##### Dynamic Analysis with Objection
 
-On a jailbroken device, you can use the iOS security assessment framework Needle to find vulnerabilities caused by the application's data storage mechanism.
+You can use the [Objection](https://github.com/sensepost/objection "Objection") runtime mobile exploration toolkit to find vulnerabilities caused by the application's data storage mechanism. Objection can be used without Jailbroken device, but it will require [Patching iOS Applications](https://github.com/sensepost/objection/wiki/Patching-iOS-Applications "Objection").
```

Comment:
```suggestion
You can use the [objection](https://github.com/sensepost/objection "objection") runtime mobile exploration toolkit to find vulnerabilities caused by the application's data storage mechanism. Objection can be used without a Jailbroken device, but it will require [patching the iOS Application](https://github.com/sensepost/objection/wiki/Patching-iOS-Applications "Objection").
```